### PR TITLE
Add `pathlib.Path` 3.14 methods

### PR DESCRIPTION
**Pull request by max-muoto**
Add new `copy`, `copytree`, and `rmtree` methods for 3.14.

https://docs.python.org/es/dev/whatsnew/3.14.html#pathlib

### CODE REVIEW

File: stdlib/pathlib.pyi [hunk 1]
```diff
@@ -159,6 +159,20 @@ class Path(PurePath):
     def lchmod(self, mode: int) -> None: ...
     def lstat(self) -> stat_result: ...
     def mkdir(self, mode: int = 0o777, parents: bool = False, exist_ok: bool = False) -> None: ...
+
+    if sys.version_info >= (3, 14):
+        def copy(self, target: StrOrBytesPath, *, follow_symlinks: bool = True, preserve_metadata: bool = False) -> None: ...
```
Comment (Akuli) [inline stdlib/pathlib.pyi:164]: Does not accept bytes (typical for pathlib):

```python
>>> pathlib.Path("python").copy(b"/home/akuli/xyz")
Traceback (most recent call last):
  File "<python-input-2>", line 1, in <module>
    pathlib.Path("python").copy(b"/home/akuli/xyz")
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "/home/akuli/sourcestuff/cpython/Lib/pathlib/_abc.py", line 816, in copy
    target = self.with_segments(target)
  File "/home/akuli/sourcestuff/cpython/Lib/pathlib/_abc.py", line 141, in with_segments
    return type(self)(*pathsegments)
           ~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/home/akuli/sourcestuff/cpython/Lib/pathlib/_local.py", line 134, in __init__
    raise TypeError(
    ...<2 lines>...
        f"not {type(path).__name__!r}")
TypeError: argument should be a str or an os.PathLike object where __fspath__ returns a str, not 'bytes'
```

```suggestion
        def copy(self, target: StrPath, *, follow_symlinks: bool = True, preserve_metadata: bool = False) -> None: ...
```

File: stdlib/pathlib.pyi [hunk 2]
```diff
@@ -159,6 +159,20 @@ class Path(PurePath):
...
+        def copytree(
+            self,
+            target: StrOrBytesPath,
```
Comment (Akuli) [inline stdlib/pathlib.pyi:167]:
```suggestion
            target: StrPath,
```

File: stdlib/pathlib.pyi [hunk 3]
```diff
@@ -236,6 +250,9 @@ class Path(PurePath):
     def replace(self, target: str | PurePath) -> Self: ...
     def resolve(self, strict: bool = False) -> Self: ...
     def rmdir(self) -> None: ...
+    if sys.version_info >= (3, 14):
+        def rmtree(self, ignore_errors: bool = False, on_error: Callable[[OSError], object] | None = None) -> None: ...
```
Comment (Akuli) [inline stdlib/pathlib.pyi:254]: I don't have this method on latest main branch of cpython.

```
$ ./python 
Python 3.14.0a0 (heads/main:5580f31c56e, Aug 10 2024, 22:50:58) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pathlib
>>> pathlib.Path.rmtree
Traceback (most recent call last):
  File "<python-input-1>", line 1, in <module>
    pathlib.Path.rmtree
AttributeError: type object 'Path' has no attribute 'rmtree'
```

Comment (max-muoto) [reply stdlib/pathlib.pyi:254]: It looks like they renamed it to `delete` since I tackled this.

Comment (max-muoto) [reply stdlib/pathlib.pyi]: https://github.com/python/typeshed/pull/12487/commits/113f2508239a63eb2ed1e67579683ee324d0ab84

**Review by Akuli — APPROVED** [review]

**Pull request merged**